3_calibrate.R

```diff
@@ -7,24 +7,14 @@ p3 <- list(
   tar_target(
     p3_nldas_glm_calibration_runs,
     {
-      # check mapping to ensure model configuration matches nml object list order
-      tar_assert_identical(p1_nldas_model_config$site_id,
-                           p1_nldas_nml_objects$morphometry$lake_name,
-                           "p1_nldas_nml_objects site id doesn't match p1_nldas_model_config site id")
-
-      # check mapping to ensure model configuration matches nml met file
-      tar_assert_identical(basename(p1_nldas_model_config$meteo_fl),
-                           p1_nldas_nml_objects$meteorology$meteo_fl,
-                           "p1_nldas_nml_objects meteo file doesn't match p1_nldas_model_config meteo file")
-
       run_glm_cal(
         nml_obj = p1_nldas_nml_objects,
         sim_dir = '3_calibrate/out',
         cal_parscale = c('cd' = 0.0001, 'sw_factor' = 0.02, 'Kw' = 0.01),
         model_config = p1_nldas_model_config,
-        optimize = TRUE)
+        calibrate = TRUE)
     },
-    # packages = c('retry','glmtools', 'GLM3r'), # not currently using `retry`
-    pattern = map(p1_nldas_model_config, p1_nldas_nml_objects))
+    pattern = map(p1_nldas_model_config)
```

Comment:
My understanding of `pattern = map(thing1, thing2)` is that it works similarly to `mapply`. In the case of `p1_nldas_model_config` it is taking the first row of the df and in `p1_nldase_nml_objects` it's taking the first object in the list. I copied this pattern from [`lake-temperature-process-models`](https://github.com/USGS-R/lake-temperature-process-models/blob/7104ca103a523b55acef15e74ffe4149f800bb08/2_run.R#L63).

I made the change you are highlighting to simplify the mapping, but more importantly, to make the mapping work at all. By adding in the subset calibration run, `p1_nldas_model_config` expands from 8 rows to 32 rows, so you can no longer maps it with `p1_nldase_nml_objects` because `p1_nldase_nml_objects` only has 8 objects in the list.

I *think**  `pattern = cross(thing1, thing2)` works like `grid.expand` - taking all unique combinations of the mapping objects ([doc link](https://books.ropensci.org/targets/dynamic.html#patterns)).